common/client.go

```diff
@@ -202,7 +202,7 @@ func (c *DatabricksClient) configureHTTPCLient() {
 	}
 }
 
-// IsAzure returns true if Azure is configured
-func (c *DatabricksClient) IsAzure() bool {
+// UsingAzureAuth returns true if Azure authentication is configured, this is not a valid check to determine if we are running on Azure
```

Comment:
Only comment I have is the rename of this makes this not so readable. I think this should be prefixed with an Is or Has. IMO Using implies that you will be returning the pointer of the parent object in method chaining. In this case it makes it seem UsingAzureAuth returns an instance of the client.

```
client.UsingAzureAuth().do(....)
```

where as being prefixed with Is or Has (preferably Is) implies that it will return a boolean value.

```
client.IsUsingAzureAuth() or client.HasUsingAzureAuth()
```